diff --git a/meson.build b/meson.build
index 429ac249..bf4ca4c5 100644
--- a/meson.build
+++ b/meson.build
@@ -54,7 +54,7 @@ endif
 
 spa_plugindir = join_paths(pipewire_libdir, spa_name)
 
-alsadatadir = join_paths(pipewire_datadir, 'alsa-card-profile', 'mixer')
+alsadatadir = join_paths(pipewire_libdir, '..', 'share', 'alsa-card-profile', 'mixer')
 
 pipewire_headers_dir = join_paths(pipewire_name, 'pipewire')
 
