diff --git a/meson.build b/meson.build
index c978370..9e82a3d 100644
--- a/meson.build
+++ b/meson.build
@@ -45,7 +45,9 @@ sdl_dep = dependency('SDL2')
 pipewire_dep = dependency('libpipewire-0.3', required: get_option('pipewire'))
 librt_dep = cppc.find_library('rt', required : get_option('pipewire'))
 
-stb_dep = dependency('stb')
+stb_dep = declare_dependency(
+  include_directories: include_directories('@stb@/include/stb')
+)
 
 wlroots_dep = dependency(
   'wlroots',
