diff --git a/usr/bin/steamos-session b/usr/bin/steamos-session
index 59b0208..9b5239a 100644
--- a/usr/bin/steamos-session
+++ b/usr/bin/steamos-session
@@ -10,37 +10,35 @@ export MANGOHUD=1
 export MANGOHUD_CONFIG="no_display,toggle_hud=F3"
 
 # Add our bin directory with the set_hd_mode and dpkg-query replacement scripts
-export PATH=/usr/share/steamos-compositor-plus/bin:${PATH}
+export PATH=@out@/share/steamos-compositor-plus/bin:${PATH}
 
-set_hd_mode.sh >> $HOME/set_hd_mode.log
+@systemd@/bin/systemd-cat -t set_hd_mode.sh
 
-export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libmodeswitch_inhibitor.so:/usr/lib/i386-linux-gnu/libmodeswitch_inhibitor.so
+export LD_LIBRARY_PATH=@modeswitchLibs@
 
 # Disable DPMS and screen blanking for now; it doesn't know about controller
 # or Steam remote control events right now
-xset -dpms
-xset s off
+@xset@/bin/xset -dpms
+@xset@/bin/xset s off
 
-if command -v start-pulseaudio-x11 > /dev/null; then
-	start-pulseaudio-x11 &
-fi
+@pulseaudio@/bin/start-pulseaudio-x11 &
 
-if systemctl is-active --quiet bluetooth && command -v bluetoothctl > /dev/null; then
-	bluetoothctl power on
+if systemctl is-active --quiet bluetooth; then
+	@bluez@/bin/bluetoothctl power on
 fi
 
 CONFIG_PATH=${XDG_CONFIG_HOME:-$HOME/.config}
 CONFIG_FILE="$CONFIG_PATH/steamos-compositor-plus"
 
-STEAMCMD="steam -tenfoot -steamos -fulldesktopres"
-COMPOSITORCMD="steamcompmgr"
+STEAMCMD="@steam@/bin/steam -tenfoot -steamos -fulldesktopres"
+COMPOSITORCMD="@out@/bin/steamcompmgr"
 if [ -f "$CONFIG_FILE" ]; then
 	source "$CONFIG_FILE"
 fi
 
 $COMPOSITORCMD &
 
-loadargb_cursor /usr/share/icons/steam/arrow.png
+@out@/bin/loadargb_cursor @out@/share/icons/steam/arrow.png
 
 # Workaround for Steam login issue while Steam client change propagates out of Beta
 touch ~/.steam/root/config/SteamAppData.vdf || true
@@ -51,7 +49,7 @@ IMAGES_DIR="$DATA_HOME/Steam/tenfoot/resource/images"
 BOOTSTRAP_IMAGE="$IMAGES_DIR/bootstrapper.jpg"
 if [ ! -f "$BOOTSTRAP_IMAGE" ]; then
 	mkdir -p "$IMAGES_DIR"
-	cp /usr/share/pixmaps/steam-bootstrapper.jpg "$BOOTSTRAP_IMAGE"
+	cp @out@/share/pixmaps/steam-bootstrapper.jpg "$BOOTSTRAP_IMAGE"
 fi
 
 if command -v steam-tweaks > /dev/null; then
diff --git a/usr/share/steamos-compositor-plus/bin/set_hd_mode.sh b/usr/share/steamos-compositor-plus/bin/set_hd_mode.sh
index cef3344..f492a18 100755
--- a/usr/share/steamos-compositor-plus/bin/set_hd_mode.sh
+++ b/usr/share/steamos-compositor-plus/bin/set_hd_mode.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!@runtimeShell@
 
 # This script attempts to set a known-good mode on a good output
 
@@ -42,13 +42,13 @@ else
 fi
 
 # First, some logging
-date
-xrandr --verbose
+@coreutils@/bin/date
+@xrandr@/bin/xrandr --verbose
 
 # List connected outputs
-ALL_OUTPUT_NAMES=$(xrandr | grep ' connected' | cut -f1 -d' ')
+ALL_OUTPUT_NAMES=$(@xrandr@/bin/xrandr | @gnugrep@/bin/grep ' connected' | @coreutils@/bin/cut -f1 -d' ')
 # Default to first connected output
-OUTPUT_NAME=$(echo $ALL_OUTPUT_NAMES | cut -f1 -d' ')
+OUTPUT_NAME=$(echo $ALL_OUTPUT_NAMES | @coreutils@/bin/cut -f1 -d' ')
 
 # If any is connected, give priority to HDMI then DP
 OUTPUT_PRIORITY="HDMI DP"
