diff --git a/tramp-gvfs.el b/tramp-gvfs.el
index fa1a90bc84..1d3dc9dfa1 100644
--- a/lisp/tramp-gvfs.el
+++ b/lisp/tramp-gvfs.el
@@ -126,5 +126,6 @@
 	     ;; for some processes.  Better we don't check.
 	     (<= emacs-major-version 25)
 	     (tramp-process-running-p "gvfs-fuse-daemon")
+             (tramp-process-running-p ".gvfsd-fuse-wrapped")
 	     (tramp-process-running-p "gvfsd-fuse"))))
   "Non-nil when GVFS is available.")
